### ISSUE TITLE
Add support to TLS

### ISSUE THREAD

**Issue by alambike** [alambike/eixo-docker#6]
If you force to connect the Eixo::Docker client with a docker remote api over https the client fails with : 
```
Uncaught exception from user code:
	malformed JSON string, neither tag, array, object, number, string or atom, at character offset 0 (before "\x{15}\x{3}\x{1}\x{0}...") at /usr/local/share/perl/5.18.2/Eixo/Rest/Request.pm line 76.
	Eixo::Rest::Request::unmarshall('Eixo::Rest::RequestSync=HASH(0x2a9cfb0)', 'HTTP::Response=HASH(0x2b416e0)') called at /usr/local/share/perl/5.18.2/Eixo/Rest/Request.pm line 38
	Eixo::Rest::Request::end('Eixo::Rest::RequestSync=HASH(0x2a9cfb0)', 'HTTP::Response=HASH(0x2b416e0)') called at /usr/local/share/perl/5.18.2/Eixo/Rest/RequestSync.pm line 42
	Eixo::Rest::RequestSync::send('Eixo::Rest::RequestSync=HASH(0x2a9cfb0)', 'LWP::UserAgent=HASH(0x2a9cf80)', 'HTTP::Request=HASH(0x2af4808)') called at /usr/local/share/perl/5.18.2/Eixo/Rest/Client.pm line 229
	Eixo::Rest::Client::__send('Eixo::Rest::Client=HASH(0x1c1d1f0)', 'HTTP::Request=HASH(0x2af4808)', 'GET_DATA', 'HASH(0x214f458)', 'PROCESS_DATA', 'HASH(0x214f4d0)', '__callback', 'CODE(0x18245c8)', 'POST_DATA', ...) called at /usr/local/share/perl/5.18.2/Eixo/Rest/Client.pm line 120

```

The **NAK ETX SOH NUL STX STX** https response handshake can't be decoded by Eixo::Rest

**Comment by alambike** [comment]
In the [docker daemon api documentation](https://docs.docker.com/reference/commandline/daemon/) suggest the use of port tcp 2375 for http connections and 2376 for https, but isn't enforced so I think we must use a flag in Eixo::Rest to specify that is a  tls connection:
```tls => 1|0```

https://github.com/alambike/eixo-rest/issues/2

**Comment by alambike** [comment]
We can also make the same process that docker client to check if must use tls connection or not.
Just checking environment vars (DOCKER_HOST and DOCKER_TLS_VERIFY):

- DOCKER_TLS_VERIFY defined =>  https
- DOCKER_TLS_VERIFY undefined => http

**Comment by alambike** [comment]
Released v1.205 with https support 

https://github.com/alambike/eixo-docker/releases/tag/1.205

Needs ```Eixo::Rest >= 1.020```

**Comment by alambike** [comment]
Released v1.205 with https support

https://github.com/alambike/eixo-docker/releases/tag/1.205

Needs Eixo::Rest >= 1.020
